File: ComputationalGeometry/ApplicationOfTriangulation/myLibraries/util/geometry/DCEL/Vertex.java
package myLibraries.util.geometry.DCEL;

/*
 * Vertex.java
 *
 * Version:
 *     $1.0$
 *
 * Revisions:
 *     $1.0$
 */

import myLibraries.util.geometry.elements.point.Vector;

/**
 * Data structure of vertex for DCEL
 *
 * @author       Xiaoyu Tongyang, or call me sora for short
 */

public class Vertex extends Vector {
    private static int IDStatic = 0;
    public HalfEdge incidentEdge;

    /**
     * constructs to create an instance of Vertex
     * */

    public Vertex( double x, double y ) {
        super( x, y, IDStatic++ );
    }

    public static
    void main( String[] args ) {
        Vertex vertex = new Vertex( 1, 1 );
        vertex.mappingID = 1;
    }
}

File: web/myLibraries/util/geometry/DCEL/Vertex.js
"use strict"

/*
 * Vertex.java
 *
 * Version:
 *     $1.0$
 *
 * Revisions:
 *     $1.0$
 */

import Vector from "../elements/point/Vector.js";

/**
 * Data structure of vertex for DCEL
 *
 * The source code in java is from my own github:
 * @see <a href=>Vertex</a>
 *
 * @author       Xiaoyu Tongyang, or call me sora for short
 */

export default class Vertex extends Vector {
    static IDStatic = 0;
    static NORMAL_COLOR = [ 136 / 255, 136 / 255, 136 / 255, 1 ];

    /**
     * constructs to create an instance of Vertex
     *
     * @param {Number} x
     * @param {Number} y
     * */

    constructor( x, y ) {
        super( x, y, Vertex._IDStatic++ );
        this.incidentEdge = null;
    }
}

